### ISSUE TITLE
Improve performance of compiled templates when getting the template name

### ISSUE THREAD

**Issue by javiereguiluz** [twigphp/Twig#4020]
I was checking some compiled Twig templates and I saw that the following is pretty common:

```php
class __TwigTemplate_da90ce2ffeefb39d5037d94ccb6fc074 extends Template
{
    // ...

    protected function doDisplay(array $context, array $blocks = [])
    {
        // ...
        $__internal_5a27a8ba21ca79b61932376b2fa922d2->enter(
            $__internal_5a27a8ba21ca79b61932376b2fa922d2_prof = new \Twig\Profiler\Profile(
                $this->getTemplateName(),
                "template",
                "@WebProfiler/Collector/events.html.twig"
            )
        );

        // ...
    }

    public function getTemplateName()
    {
        return "some/path/my_template.html.twig";
    }

    // ...
}
```

Calling a method is ~1.70x slower than accessing a property (see benchmark: https://3v4l.org/UPsnU) and in some templates, the code calls `$this->getTemplateName()` multiple times.

So, I wonder if compiled templates could be like this instead:

```diff
 <?php
 
 class __TwigTemplate_da90ce2ffeefb39d5037d94ccb6fc074 extends Template
 {
+    private string $templateName = 'some/path/my_template.html.twig';
     // ...
 
     protected function doDisplay(array $context, array $blocks = [])
     {
         // ...
         $__internal_5a27a8ba21ca79b61932376b2fa922d2->enter(
             $__internal_5a27a8ba21ca79b61932376b2fa922d2_prof = new \Twig\Profiler\Profile(
-                $this->getTemplateName(),
+                $this->templateName,
                 "template",
                 "@WebProfiler/Collector/events.html.twig"
             )
         );
 
         // ...
     }
 
     public function getTemplateName()
     {
-        return "some/path/my_template.html.twig";
+        return $this->templateName;
     }
 
     // ...
 }
```

If we agree on this change, we should think of the best way of implementing this for performance: typed private property, untyped private property, private static property, private constant, etc.

**Comment by nicolas-grekas** [comment]
I'm not sure this would ever make a difference. Can you find a twig benchmark that shows when this would be beneficial?

**Comment by javiereguiluz** [comment]
I inspected the `var/.../twig/` dir of a medium-sized Symfony app. The `$this->getTemplateName()` instruction is used 1,454 times. The template with the most calls uses it 102 times.

If you mean that this won't make a difference because the gain is small ... it's true that it's a small gain. But, in my opinion, in a mature project like Twig you can't make a small change and improve performance 1%. The way you do that is by making 10 small changes that improve performance 0.1% each.

**Comment by stof** [comment]
The question is still how much this improves performance.

**Comment by lyrixx** [comment]
I check in production, and indeed, there are some all to `getTemplateName()`.
But it's always wrapped in `new \Twig\Profiler\Profile()`.

So I checked the code, and it seems it's the only place where we use this method in compiled cache.

But we don't have the profiler enabled in production.

It comes from the sentry profiler. I tried to disable it and calls are gone.

So if we change something, **it'll be useful only in dev** (or special cases)
